### ISSUE TITLE
"Replacable" blocks

### ISSUE THREAD

**Issue by microsoftsam** [Microsoft/pxt-blockly#82]
In PXT we would like to create new "ghost" blocks that have a bit of text in them. These blocks would fit inside other C shaped blocks, just like normal blocks, but they can't be dragged out of position. The idea is that you would replace these blocks with other blocks. 

The closest thing to this is shadow blocks. But instead of just using it as the VALUE of a block, we would use it inside a STACK.
I've tried doing it by just defining the STACK inside of a block, and using a shadow block rather than a real block, but the issue I run into is when I drag something above it

Thoughts on the best way to tackle this? @rachel-fenichel 

I think shadow blocks make sense, we just need to exclude them from drag / insertion marker logic. 

Here's a gif of the desired outcome, and the issues with insertion: 
![shadowcommandblocks](https://user-images.githubusercontent.com/16690124/32465696-2f952728-c2f9-11e7-935f-8607df02f3ca.gif)

**Comment by microsoftsam** [comment]
<img width="324" alt="screen shot 2017-11-06 at 1 57 31 pm" src="https://user-images.githubusercontent.com/16690124/32466046-798231e0-c2fa-11e7-896e-a11c88478c22.png">

**Comment by microsoftsam** [comment]
<img width="393" alt="screen shot 2017-11-06 at 1 59 53 pm" src="https://user-images.githubusercontent.com/16690124/32466149-cab871a0-c2fa-11e7-81e3-f7798c6bba53.png">

**Comment by rachel-fenichel** [comment]
I agree that shadow blocks make sense here.  It looks like there's just a problem with the insertion marker code for handling shadow statements, probably in `block_dragger.js` or `insertion_marker_manager.js`.

To double-check: this is the behaviour you would want, right?
![3aec8343-f83e-47e8-8558-214194535e1a](https://user-images.githubusercontent.com/13686399/32467700-7b76ec2e-c300-11e7-89dc-bee6fe78cfcf.gif)

**Comment by microsoftsam** [comment]
Yes that's correct. Okay cool I'll look into that.
